### PR TITLE
core: move to sdbus-cpp2

### PR DESCRIPTION
**Pull request by vaxerski**
Needs testing

@fufexan ci

### CODE REVIEW

**Comment by VirtCode** [comment]
Things like `lock_cmd`, `before_sleep_cmd` and `after_sleep_cmd` dont seem to work anymore. Probably caused by these error messages which are printed:
```
...
[LOG] wayland done, registering dbus
[WARN] Couldn't connect to logind service ([org.freedesktop.DBus.Error.ServiceUnknown] The name is not activatable)
[LOG] Using dbus path
[WARN] Couldn't retrieve current systemd inhibits ([org.freedesktop.DBus.Error.ServiceUnknown] The name is not activatable)
[TRACE] [core] got poll event
...
```
main (for reference):
```
...
[LOG] wayland done, registering dbus
[LOG] Using dbus path /org/freedesktop/login1/session/_33
[TRACE] [core] got poll event
...
```

Otherwise listeners and their inhibition seems to work (tested with [wayland idle inhibitor](https://github.com/stwa/wayland-idle-inhibitor))

**Comment by vaxerski** [comment]
@VirtCode try now

**Comment by VirtCode** [comment]
The connection seems to do something now:
```
[LOG] wayland done, registering dbus
[LOG] Using dbus path /org/freedesktop/login1/session/_33
```

However, the before_sleep commands etc. are still not working. Also, the normal listener don't seem to work anymore. It's odd that there are no longer any `[TRACE] [core] got poll event` printed.

**Comment by vaxerski** [comment]
piss of shit, I'll fix it later, not at pc rn

**Comment by vaxerski** [comment]
@andresilva you committed that feature, any ideas?

File: src/core/Hypridle.cpp [hunk 1]
```diff
@@ -531,21 +517,28 @@ void CHypridle::setupDBUS() {
...
-                    obj->registerMethod("org.freedesktop.ScreenSaver", "Inhibit", "ss", "u", [&](sdbus::MethodCall c) { handleDbusScreensaver(c, true); });
-                    obj->registerMethod("org.freedesktop.ScreenSaver", "UnInhibit", "u", "", [&](sdbus::MethodCall c) { handleDbusScreensaver(c, false); });
-                    obj->finishRegistration();
+                    auto obj = sdbus::createObject(*m_sDBUSState.screenSaverServiceConnection, sdbus::ObjectPath{path});
+
+                    obj->addVTable(sdbus::registerMethod("Inhibit").implementedAs([object = obj.get()](std::string s1, std::string s2) {
+                           return handleDbusScreensaver(s1, s2, 0, true, object->getCurrentlyProcessedMessage().getSender());
+                       }),
+                                   sdbus::registerMethod("UnInhibit").implementedAs([object = obj.get()](uint32_t c) {
+                                       handleDbusScreensaver("", "", c, true, object->getCurrentlyProcessedMessage().getSender());
```
Comment (andresilva) [inline src/core/Hypridle.cpp:530]:
```suggestion
                                       handleDbusScreensaver("", "", c, false, object->getCurrentlyProcessedMessage().getSender());
```

Comment (vaxerski) [reply src/core/Hypridle.cpp:530]: oops. truly.

**Comment by vaxerski** [comment]
thanks bae

**Comment by VirtCode** [comment]
yeah, also saw this yesterday and tried to patch it out locally, but it doesn't fix the issue. Can still reproduce it on 0.1.4.

For me it looks like the second callback (`UnInhibit`) just never gets called.

**Comment by vaxerski** [comment]
open a new issue

**Pull request merged**